source/magpie/dataframe.d

```diff
@@ -1034,15 +1134,28 @@ public:
         }
         else
         {
-            static foreach(i; 0 .. RowType.length)
+            void assignAux(ptrdiff_t si = -1)(size_t ri = 0) @property
             {
-                if(i == pos)
+                static if(si > -1)
+                {
+                    alias i = si;
+                    enum size_t typepos = i;
+                }
+                else
+                {
+                    size_t i = ri;
+                    enum size_t typepos = 0;
+                }
+
+                static if(is(toArr!(RowType[typepos]) == U[0]))
                 {
-                    static if(is(FrameType[i] == U[0]))
-                        foreach(j; 0 .. (rows > values[0].length)? values[0].length: rows)
-                            data[i][j] = values[0][j];
+                    foreach(j; 0 .. (rows > values[0].length)? values[0].length: rows)
+                        data[i][j] = values[0][j];
                 }
             }
+            
+            mixin auxDispatch!(assignAux);
+            auxDispatch(pos);
```

Comment:
the line count increase for doing this here is quite large and the entire point of it was to increase readability which I don't think this does.